pyccel/parser/semantic.py

```diff
@@ -724,7 +724,7 @@ def get_type_description(self, var, include_rank = True):
         """
         dtype = var.dtype
         prec  = get_final_precision(var)
-        descr = f'{dtype}(kind={prec})'
+        descr = f'{dtype}{prec * 8 if prec else ""}'
```

Comment:
For `complex` type will output the wrong precision. I don't exactly know why but I found [this](https://gcc.gnu.org/onlinedocs/gfortran/KIND-Type-Parameters.html#:~:text=The%20KIND%20value%20matches%20the%20storage%20size%20in%20bytes%2C%20except%20for%20COMPLEX%20where%20the%20storage%20size%20is%20twice%20as%20much)

you can see the issue with this code:
```python
def a(arg: 'complex128'):
    pass
a(4)
```
this will output 
```
Argument 1 : 4 (int64), passed to function a is incompatible (expected complex64)
```
but I think we should expect to see:
```
Argument 1 : 4 (int64), passed to function a is incompatible (expected complex128)
```

Reply:
See #1167 and #1134 for details